client/src/app/pages/applications/analysis-wizard/set-targets.tsx

```diff
@@ -168,25 +212,51 @@ export const SetTargets: React.FC<SetTargetsProps> = ({ applications }) => {
             title={t("wizard.label.skipTargets")}
           />
         )}
-
-      <Gallery hasGutter>
-        {targetsToRender.map((target) => (
-          <GalleryItem key={target.id}>
-            <TargetCard
-              readOnly
-              item={target}
-              cardSelected={selectedTargets.some(({ id }) => id === target.id)}
-              onSelectedCardTargetChange={(selectedTarget) => {
-                handleOnSelectedCardTargetChange(selectedTarget);
-              }}
-              onCardClick={(isSelecting, selectedLabelName, target) => {
-                handleOnCardClick(isSelecting, selectedLabelName, target);
-              }}
-              formLabels={formLabels}
-            />
-          </GalleryItem>
-        ))}
-      </Gallery>
+      <ConditionalTableBody
+        isLoading={isLoading}
+        isError={isError}
+        isNoData={targets.length === 0}
+        numRenderedColumns={1}
+      >
+        <Gallery hasGutter>
+          {currentPageItems.map((target) => (
+            <GalleryItem key={target.id}>
+              <TargetCard
+                readOnly
+                item={target}
+                cardSelected={selectedTargets.some(
+                  ({ id }) => id === target.id
+                )}
+                onSelectedCardTargetChange={(selectedTarget) => {
+                  handleOnSelectedCardTargetChange(selectedTarget);
+                }}
+                onCardClick={(isSelecting, selectedLabelName, target) => {
+                  handleOnCardClick(isSelecting, selectedLabelName, target);
+                }}
+                formLabels={formLabels}
+              />
+            </GalleryItem>
+          ))}
+        </Gallery>
+      </ConditionalTableBody>
     </Form>
   );
 };
+
+export const SetTargets: React.FC<SetTargetsProps> = ({ applications }) => {
+  // pre-fetch data but leave error handling to the real page
+  const { isLoading, applicationProviders } = useEnhancedTargets(applications);
+  if (isLoading) {
+    return (
+      <Bullseye>
+        <Spinner size="xl" />
+      </Bullseye>
```

Comment:
Ideally this would be one of those fancy skeleton previews (https://www.patternfly.org/components/skeleton)

Reply:
IMHO the panel is too small - skeletons look the best on full page.

Reply:
Good point!